common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -19,74 +19,102 @@
-                root.get("conceptName"), criteriaBuilder.literal(query));
-            queryPredicates.add(criteriaBuilder.greaterThan(matchExp, 0.0));
-            queryPredicates.add(criteriaBuilder.equal(root.get("conceptCode"),
-                criteriaBuilder.literal(query)));
-            try {
-              long conceptId = Long.parseLong(query);
-              queryPredicates.add(criteriaBuilder.equal(root.get("conceptId"),
-                  criteriaBuilder.literal(conceptId)));
-            } catch (NumberFormatException e) {
-              // Not a long, don't try to match it to a concept ID.
-            }
-            predicates.add(criteriaBuilder.or(queryPredicates.toArray(new Predicate[0])));
-
-            // Optionally filter on standard concept, vocabulary ID, domain ID
-            if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_CONCEPTS)) {
-                predicates.add(criteriaBuilder.equal(root.get("standardConcept"),
-                    criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
-            } else if (standardConceptFilter.equals(StandardConceptFilter.NON_STANDARD_CONCEPTS)) {
-              List<Predicate> standardConceptPredicates = new ArrayList<>();
-              standardConceptPredicates.add(criteriaBuilder.isNull(root.get("standardConcept")));
-              standardConceptPredicates.add(criteriaBuilder.notEqual(root.get("standardConcept"),
-                  criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
-              predicates.add(criteriaBuilder.or(
-                  standardConceptPredicates.toArray(new Predicate[0])));
-            }
-            if (vocabularyIds != null) {
-              predicates.add(root.get("vocabularyId").in(vocabularyIds));
+    public enum StandardConceptFilter {
+        ALL_CONCEPTS,
+        STANDARD_CONCEPTS,
+        NON_STANDARD_CONCEPTS
+    }
+
+    @PersistenceContext(unitName = "cdr")
+    private EntityManager entityManager;
+
+    public ConceptService() {
+    }
+
+    // Used for tests
+    public ConceptService(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public static String modifyMultipleMatchKeyword(String query){
+        // This function modifies the keyword to match all the words if multiple words are present(by adding + before each word to indicate match that matching each word is essential)
+        String[] keywords = query.split("[,+\\s+]");
+        for(int i = 0;i < keywords.length;i++){
+            String key = keywords[i];
+            if(key.length() < 3){
+                key = "\""+key+"\"";
+                keywords[i] = key;
             }
-            if (domainIds != null) {
-              predicates.add(root.get("domainId").in(domainIds));
+        }
+
+        String query2= "";
+        for(String key: keywords){
+            if(query2.isEmpty()){
+                query2 = "+"+key;
+            }else if(key.contains("\"")){
+                query2 = query2+key;
```

Comment:
Going to repeat my comment from before:

So if I type in:

hello "foo bar" baz

This will I believe produce:

+hello"foo bar"+baz

Or if I put in:

dog v dog

we'll wind up with +dog"v"+dog?

Is that correct? Is that what MATCH expects?

Please add some tests around this.

FWIW, I'd be perfectly happy if we didn't special-case the < 3 character words and let MySQL ignore them if it wants to. I don't think filtering on strings that short is that useful.

Reply:
Ohh yeah. We might ignore small words. 
The problem if we do not put in quotes is, match does not care to consider words that come after the short ones. For "dog v dog", match filters only on the first dog and does not care about the second dog at all.